test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/YearTest.scala

```diff
@@ -0,0 +1,329 @@
+    expectThrows(classOf[DateTimeException], min.minusYears(Long.MinValue))
+    expectThrows(classOf[DateTimeException], max.minusYears(Long.MaxValue))
+  }
+
+  @Test def adjustInto(): Unit = {
+    val aDate = LocalDate.of(2015, 1, 1)
+    assertEquals(LocalDate.of(0, 1, 1), lastBCYear.adjustInto(aDate))
+  }
+
+  @Test def until(): Unit = {
+    for {
+      t <- samples
+      unit <- ChronoUnit.values() if isSupported(unit)
+    } {
+      assertEquals(0L, t.until(t, unit))
+    }
+
+    assertEquals(1999999998L, min.until(max, YEARS))
+    assertEquals(-1999999998L, max.until(min, YEARS))
+    assertEquals(1L, lastBCYear.until(firstACYear, YEARS))
+
+    val otherLeapYear: Year = {
+      var y = leapYear.getValue + 1
+      while (!Year.isLeap(y)) y += 1
+      Year.of(y)
+    }
+    assertEquals(4L, leapYear.until(otherLeapYear, YEARS))
+
+    assertEquals(10L, lastBCYear.until(Year.of(10), YEARS))
+    assertEquals(10L, firstACYear.until(Year.of(11), YEARS))
+    assertEquals(1L, lastBCYear.until(Year.of(10), DECADES))
+    assertEquals(1L, firstACYear.until(Year.of(11), DECADES))
+    assertEquals(0L, lastBCYear.until(Year.of(10), CENTURIES))
+    assertEquals(0L, firstACYear.until(Year.of(11), CENTURIES))
+    assertEquals(0L, lastBCYear.until(Year.of(10), MILLENNIA))
+    assertEquals(0L, firstACYear.until(Year.of(11), MILLENNIA))
+
+    assertEquals(100L, lastBCYear.until(Year.of(100), YEARS))
+    assertEquals(100L, firstACYear.until(Year.of(101), YEARS))
+    assertEquals(10L, lastBCYear.until(Year.of(100), DECADES))
+    assertEquals(10L, firstACYear.until(Year.of(101), DECADES))
+    assertEquals(1L, lastBCYear.until(Year.of(100), CENTURIES))
+    assertEquals(1L, firstACYear.until(Year.of(101), CENTURIES))
+    assertEquals(0L, lastBCYear.until(Year.of(100), MILLENNIA))
+    assertEquals(0L, firstACYear.until(Year.of(101), MILLENNIA))
+
+    assertEquals(1000L, lastBCYear.until(Year.of(1000), YEARS))
+    assertEquals(1000L, firstACYear.until(Year.of(1001), YEARS))
+    assertEquals(100L, lastBCYear.until(Year.of(1000), DECADES))
+    assertEquals(100L, firstACYear.until(Year.of(1001), DECADES))
+    assertEquals(10L, lastBCYear.until(Year.of(1000), CENTURIES))
+    assertEquals(10L, firstACYear.until(Year.of(1001), CENTURIES))
+    assertEquals(1L, lastBCYear.until(Year.of(1000), MILLENNIA))
+    assertEquals(1L, firstACYear.until(Year.of(1001), MILLENNIA))
+
+    assertEquals(1L, lastBCYear.until(firstACYear, ERAS))
+    assertEquals(0L, firstACYear.until(Year.of(100), ERAS))
+    assertEquals(0L, lastBCYear.until(Year.of(-100), ERAS))
+  }
+
+  @Test def atDay(): Unit = {
+    for (t <- samples) {
+      assertEquals(LocalDate.of(t.getValue, 1, 1), t.atDay(1))
+      assertEquals(LocalDate.of(t.getValue, 12, 31), t.atDay(t.length()))
+    }
+
+    assertEquals(LocalDate.of(leapYear.getValue, 2, 29), leapYear.atDay(31 + 29))
+    assertEquals(LocalDate.of(leapYear.getValue, 3, 1), leapYear.atDay(31 + 30))
```

Comment:
Add tests for `leapYear.atDay(366)`, `leapYear.atDay(367)`, `nonLeapYear.atDay(365)`, `nonLeapYear.atDay(366)` and `.atDay(-1)`.